bin/MindeeCLICommand.php

```diff
@@ -0,0 +1,336 @@
+<?php
+
+namespace Mindee\CLI;
+
+require __DIR__ . '/../vendor/autoload.php';
+require __DIR__ . '/../src/version.php';
+
+use Mindee\Client;
+use Mindee\Error\MindeeHttpException;
+use Mindee\Input\PageOptions;
+use Mindee\Input\PredictMethodOptions;
+use Mindee\Input\PredictOptions;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use const Mindee\Input\KEEP_ONLY;
+use const Mindee\Input\REMOVE;
+use const Mindee\VERSION;
+
+const JSON_PRINT_RECURSION_DEPTH = 20;
+
+class MindeeCLICommand extends Command
+{
+    /**
+     * @var array $documentList Array of document configurations.
+     */
+    private array $documentList;
+    /**
+     * @var array $acceptableDocuments Array of acceptable documents.
+     */
+    private array $acceptableDocuments;
+
+    /**
+     * @param array $documentList Array of document configurations.
+     */
+    public function __construct(array $documentList)
+    {
+        $this->documentList = $documentList;
+
+        $this->acceptableDocuments = [];
+        foreach ($this->documentList as $documentName => $document) {
+            $this->acceptableDocuments[] = $documentName;
+        }
+        parent::__construct('mindee');
+    }
+
+    /**
+     * @param string|null $product Selected product, for customisation of the help section.
+     * @return string
+     */
+    protected function formatHelp(string $product = null): string
+    {
+        $helpCondensed = "";
+        if (!$product) {
+            $helpCondensed = "Mindee Command-Line interface.
+Usage:
+  mindee [options] [--] <product> <method> <file_path_or_url>
+
+Available products:";
+            foreach ($this->documentList as $documentName => $document) {
+                $helpCondensed .= "\n  " . str_pad($documentName, 65 - strlen($document->help)) . $document->help;
+            }
+        } else {
+            $helpCondensed .= $this->documentList[$product]->help;
+        }
+        return $helpCondensed;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('Mindee')
+            ->setDescription('Mindee client.')
+            ->addArgument(
+                'product',
+                InputArgument::REQUIRED,
+                'Specify which product to use. Available products are :' . implode("\n  ", $this->acceptableDocuments)
+            )
+            ->addArgument(
+                'file_path_or_url',
+                InputArgument::REQUIRED,
+                'Path or URL of the file to be processed.'
+            );
+
+        $this->configureMainOptions();
+        $this->configureCustomOptions();
+    }
+
+    private function configureMainOptions()
+    {
+        $this->addOption(
+            'async_polling',
+            'A',
```

Comment:
It's also possible to simply have an option `--async` with no short version.
This is how it's done in the .NET lib for example.

Reply:
Uhm, I kind of liked having a shorthand for that. Can I keep it? 🥹